packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx

```diff
@@ -0,0 +1,192 @@
+import { useEffect, useState } from 'react'
+
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalTitle,
+  SegmentedControl
+} from '@audius/stems'
+
+import {
+  FEATURE_FLAG_OVERRIDE_KEY,
+  OverrideSetting
+} from 'common/hooks/useFeatureFlag'
+import { useModalState } from 'common/hooks/useModalState'
+import { FeatureFlags } from 'common/services/remote-config'
+import zIndex from 'utils/zIndex'
+
+import styles from './FeatureFlagOverrideModal.module.css'
+
+const messages = {
+  title: 'Feature Flag Override Settings'
+}
+
+const getOverrideSetting = (flag: FeatureFlags) =>
+  localStorage.getItem(
+    `${FEATURE_FLAG_OVERRIDE_KEY}:${flag}`
+  ) as OverrideSetting
+
+const setOverrideSetting = (flag: FeatureFlags, val: OverrideSetting) => {
+  const flagKey = `${FEATURE_FLAG_OVERRIDE_KEY}:${flag}`
+  if (val === null) return localStorage.removeItem(flagKey)
+  localStorage.setItem(flagKey, val)
+}
+
+export const FeatureFlagOverrideModal = () => {
+  const [isOpen, setIsOpen] = useModalState('FeatureFlagOverride')
+
+  const [challengeRewardsUiSetting, setChallengeRewardsUiSetting] = useState(
```

Comment:
Is there any way we can map over the the FeatureFlag enum to create the initial state & info map? Would be ideal not to have to change another file when adding/removing flags.

Something like this maybe:

```ts
const [flagOverrideState, setFlagOverrideState] = useState(
  mapValues(FeatureFlags, (flag) => getOverrideSetting(flag))
)
```

Reply:
i was looking for a way to do this. I'll try this out and see if I can get it to work 👍